### PR TITLE
rework internal events

### PR DESCRIPTION
**Pull request by timc1**
Closes https://github.com/timc1/kbar/issues/25.

- Closing `kbar` using keyboard shortcuts; e.g. `ESC`/`cmd+k` should not scroll to the bottom of the page on Safari
- Prevent the underlying document from scrolling when kbar is present
- Cleans up `InternalEvents`, separating varying logic into separate components

### CODE REVIEW

File: src/InternalKeyboardEvents.tsx [hunk 1]
```diff
@@ -4,10 +4,23 @@ import useKBar from "./useKBar";
...
 
-export default function InternalKeyboardEvents() {
-  const { query, options, visualState, actions } = useKBar((state) => ({
+export default function InternalEvents() {
+  return (
+    <>
+      <ToggleHandler />
+      <ShortcutsHandler />
+      <FocusHandler />
+      <DocumentLock />
```
Comment (timc1) [inline src/InternalKeyboardEvents.tsx:13]: Have some separation of concerns for each type of event, not fully set on this but seem quite nice & simple.

Comment (tommoor) [reply src/InternalKeyboardEvents.tsx:13]: Interesting pattern, if all of these `return null` and reason not to use hook-style?

Comment (timc1) [reply src/InternalKeyboardEvents.tsx:13]: Can totally convert these to hooks style, will do. Do you know of a better pattern we can use to register these different event listeners?

Comment (tommoor) [reply src/InternalKeyboardEvents.tsx:13]: Not particularly, I'd _consider_ depending on another library for the keyboard shortcuts functionality though as you go forward – I feel like there are bunch of edge cases to consider that don't need to be dealt with in kbar.

Comment (timc1) [reply src/InternalKeyboardEvents.tsx]: Going to convert these to hooks style for now and get this in as a first step – will further look into external libs for keyboard shortcuts, want to really get this right 💯

**Pull request merged**